modules/calcite/src/main/java/org/apache/ignite/internal/processors/query/calcite/prepare/IgnitePlanner.java

```diff
@@ -221,7 +225,20 @@ public ValidationResult validateAndGetTypeMetadata(SqlNode sqlNode) {
     @Override public RelRoot rel(SqlNode sql) {
         SqlToRelConverter sqlToRelConverter = sqlToRelConverter(validator(), catalogReader, sqlToRelConverterCfg);
         RelRoot root = sqlToRelConverter.convertQuery(sql, false, true);
-        root = root.withRel(sqlToRelConverter.decorrelate(sql, root.rel));
+        RelNode rootRel = root.rel;
```

Comment:
Why did you do this change?

Reply:
cause with .withExpand(false) correlates are not deduplicated, there are discussion on calcite dev list, i show you.

Reply:
fill the ticket and pr https://issues.apache.org/jira/browse/CALCITE-4673